include/xtypes/DynamicDataImpl.hpp

```diff
@@ -110,6 +110,925 @@ inline std::string ReadableDynamicDataRef::to_string() const
+           type_.kind() == TypeKind::INT_16_TYPE || type_.kind() == TypeKind::UINT_16_TYPE ||
+           type_.kind() == TypeKind::INT_32_TYPE || type_.kind() == TypeKind::UINT_32_TYPE ||
+           type_.kind() == TypeKind::INT_64_TYPE || type_.kind() == TypeKind::UINT_64_TYPE ||
+           type_.kind() == TypeKind::FLOAT_32_TYPE || type_.kind() == TypeKind::FLOAT_64_TYPE ||
+           type_.kind() == TypeKind::FLOAT_128_TYPE);
+
+    switch(type_.kind())
+    {
+        case TypeKind::INT_8_TYPE:
+            {
+                DynamicData result(primitive_type<int8_t>());
+                result = -(*this);
+                return result;
+            }
+        case TypeKind::UINT_8_TYPE:
+            {
+                DynamicData result(primitive_type<uint8_t>());
+                result = -(*this);
+                return result;
+            }
+        case TypeKind::INT_16_TYPE:
+            {
+                DynamicData result(primitive_type<int16_t>());
+                result = -(*this);
+                return result;
+            }
+        case TypeKind::UINT_16_TYPE:
+            {
+                DynamicData result(primitive_type<uint16_t>());
+                result = -(*this);
+                return result;
+            }
+        case TypeKind::INT_32_TYPE:
+            {
+                DynamicData result(primitive_type<int32_t>());
+                result = -(*this);
+                return result;
+            }
+        case TypeKind::UINT_32_TYPE:
+            {
+                DynamicData result(primitive_type<uint32_t>());
+                result = -(*this);
+                return result;
+            }
+        case TypeKind::INT_64_TYPE:
+            {
+                DynamicData result(primitive_type<int64_t>());
+                result = -(*this);
+                return result;
+            }
+        case TypeKind::UINT_64_TYPE:
+            {
+                DynamicData result(primitive_type<uint64_t>());
+                result = -(*this);
+                return result;
+            }
+        case TypeKind::FLOAT_32_TYPE:
+            {
+                DynamicData result(primitive_type<float>());
+                result = -(*this);
+                return result;
+            }
+        case TypeKind::FLOAT_64_TYPE:
+            {
+                DynamicData result(primitive_type<double>());
+                result = -(*this);
+                return result;
+            }
+        case TypeKind::FLOAT_128_TYPE:
+            {
+                DynamicData result(primitive_type<long double>());
+                result = -(*this);
+                return result;
+            }
+        default:
+            break;
+    }
+
+    return *this;
+}
+
+inline DynamicData DynamicData::operator * (const ReadableDynamicDataRef& other) const
+{
+    switch(type_.kind())
+    {
+        case TypeKind::INT_8_TYPE:
+            {
+                int8_t lho = *this;
```

Comment:
A macro for this 5 lines in the top of the file, can help you to reduce and maintain all the code easily.